### PR TITLE
WIP: Add cell to CellCache

### PR DESCRIPTION
**Pull request by KnutAM**
Just to ensure that this can be done in a non-breaking way after #798 is merged. 

Did not see any performance degradation on the porous media example where the cell type is `Union{Quadrilateral, Triangle}`, so the approach in this PR does look ok AFAIU.

No need to merge now - just wanted to prevent any issues after 1.0 release..

### CODE REVIEW

**Comment by termi-official** [comment]
Can you ELI5 why the celltype should be in the CellCache (and why it should be a union)? 

Having a union might be an issue for the GPU stuff. We are currently working towards making CellCache type stable for the GPU (xref https://github.com/Ferrite-FEM/Ferrite.jl/issues/1049#issuecomment-2298441315).

**Comment by KnutAM** [comment]
Needed to support reinit! for vector interpolations using the CellIterator. 

Union: the grid has a Union, the cellCache not, but the constructor should be type unstable. Iteration is not.

**Comment by termi-official** [comment]
Thanks!

> Needed to support reinit! for vector interpolations using the CellIterator.

`reinit!` acts on  some FEValues and you can obtain the information from the geometric interpolation, right?

**Comment by KnutAM** [comment]
> `reinit!` acts on some FEValues and you can obtain the information from the geometric interpolation, right?

The actual cell is required to determine the global direction of faces or edges based on its node ids during reinit of FEValues with non-identity mappings.

**Comment by fredrikekre** [comment]
Couldn't the cell be looked up in `reinit!` instead? We have the cellid

**Comment by KnutAM** [comment]
> Couldn't the cell be looked up in reinit! instead? We have the cellid

Yes, would be possible. But that felt like a split design, to partly reinit the cell cache, and then do more lookup based on the grid in the cache during reinit of the fe values. 

And also not sure if there are performance pitfalls regarding type stability here, especially for GPU if a Union isn't allowed during `reinit!`, but haven't checked/thought carefully about this yet.

**Comment by fredrikekre** [comment]
We should not consider type params as part of the API so this is not breaking.
